### PR TITLE
Simplify command application

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: src/audio/soundtrack.rs [hunk 1]
```diff
@@ -22,29 +26,35 @@ impl Command for PlaySoundtrack {
...
+}
 
-        let soundtrack_key = match &self {
-            PlaySoundtrack::Key(key) => key.clone(),
-            PlaySoundtrack::Disable => return,
-        };
+fn apply_play_soundstrack(
+    In(play_soundtrack): In<PlaySoundtrack>,
+    mut commands: Commands,
+    soundtracks: Query<Entity, With<IsSoundtrack>>,
```
Comment (benfrankel) [inline src/audio/soundtrack.rs:36]: I know we aren't being consistent about this in other places already, but: Do we want to use the `soundtrack_query` or `soundtracks` naming convention for queries? It would be nice to make this consistent in a separate PR.

File: src/audio/sound_effects.rs [hunk 2]
```diff
@@ -61,6 +61,8 @@ struct PlaySoundEffect {
 
 impl Command for PlaySoundEffect {
     fn apply(self, world: &mut World) {
+        // If you need more complex behavior, use `world.run_system_once_with`,
+        // as demonstrated for `PlaySoundtrack`.
```
Comment (benfrankel) [inline src/audio/sound_effects.rs:65]:
```suggestion
        // as demonstrated with `PlaySoundtrack`.
```
I'm leaning that this reads slightly better, but not 100%. Can ignore this.

**Review by benfrankel — COMMENTED** [review]
No changes required, but want responses before approving.

File: src/audio/soundtrack.rs (hunk 1 above)
Comment (janhohenheim) [reply src/audio/soundtrack.rs:36]: Good point.
image
image

`_query` wins in terms of consistency. I'll change it in this PR.

Comment (janhohenheim) [reply src/audio/soundtrack.rs:36]: (I personally use `q_soundtrack`, but I'm not even putting that on the table as I'm sure it's too controversial)

Comment (janhohenheim) [reply src/audio/soundtrack.rs]: [`a155a1e` (#239)](https://github.com/TheBevyFlock/bevy_quickstart/pull/239/commits/a155a1ecf707a356e8c2bd47941b7f17a2a1f68c)

File: src/audio/soundtrack.rs [hunk 3]
```diff
@@ -22,29 +26,35 @@ impl Command for PlaySoundtrack {
     /// This command will despawn the current soundtrack, then spawn a new one
     /// if necessary.
     fn apply(self, world: &mut World) {
-        let mut soundtrack_query = world.query_filtered::<Entity, With<IsSoundtrack>>();
-        let soundtracks: Vec<_> = soundtrack_query.iter(world).collect();
-        for entity in soundtracks {
-            world.entity_mut(entity).despawn_recursive();
-        }
+        world.run_system_once_with(self, apply_play_soundstrack);
```
Comment (benfrankel) [inline src/audio/soundtrack.rs:29]:
```suggestion
        world.run_system_once_with(self, apply_play_soundtrack);
```

File: src/audio/soundtrack.rs [hunk 4]
```diff
@@ -22,29 +26,35 @@ impl Command for PlaySoundtrack {
...
+    }
+}
 
-        let soundtrack_key = match &self {
-            PlaySoundtrack::Key(key) => key.clone(),
-            PlaySoundtrack::Disable => return,
-        };
+fn apply_play_soundstrack(
```
Comment (benfrankel) [inline src/audio/soundtrack.rs:33]:
```suggestion
fn apply_play_soundtrack(
```

**Review by benfrankel — CHANGES_REQUESTED** [review]
just noticed typo, other than that approved

**Review by benfrankel — APPROVED** [review]
